libs/community/langchain_community/vectorstores/falkordb_vector.py

```diff
@@ -0,0 +1,1684 @@
+    """Check if the values are not None or empty string"""
+    for prop, value in zip(props, values):
+        if not value:
+            raise ValueError(f"Parameter `{prop}` must not be None or empty string")
+
+
+
+
+
+
+
+class FalkorDBVector(VectorStore):
+    """`FalkorDB` vector index.
+
+    To use, you should have the ``falkordb`` python package installed
+
+    Args:
+        host: FalkorDB host
+        port: FalkorDB port
+        username: Optionally provide your username 
+                  details if you are connecting to a
+                  FalkorDB Cloud database instance
+        password: Optionally provide your password
+                  details if you are connecting to a
+                  FalkorDB Cloud database instance
+        embedding: Any embedding function implementing
+                `langchain.embeddings.base.Embeddings` interface.
+        distance_strategy The distance strategy to use. (default: "EUCLIDEAN")
+        pre_delete_collection: If True, will delete existing data if it exists.
+                (default: False). Useful for testing.
+
+
+    Example:
+        .. code-block:: python
+
+            from langchain_community.vectorstores.falkordb_vector import FalkorDBVector
+            from langchain_community.embeddings.openai import OpenAIEmbeddings
+
+            host="localhost"
+            port=6379
+            embeddings=OpenAIEmbeddings()
+            vectorstore = FalkorDBVector.from_documents(
+                embedding=embeddings,
+                documents=docs,
+                host=host,
+                port=port,  
+            )
+
+
+        
+
+    """
+    def __init__(
+        self,
+        embedding: Embeddings,
+        *,
+        search_type: SearchType = SearchType.VECTOR,
+        username: Optional[str] = None,
+        password: Optional[str] = None,
+        host: str = "localhost",
+        port: int = 6379,
+        distance_strategy: DistanceStrategy = DEFAULT_DISTANCE_STRATEGY,
+        database: str = generate_random_string(4),
+        node_label: str = "Chunk",
+        relation_type: str = "",
+        embedding_node_property: str = "embedding",
+        text_node_property: str = "text",
+        embedding_dimension: Optional[int] = None,
+        retrieval_query: str = "",
+        index_type: IndexType = DEFAULT_INDEX_TYPE,
+        graph: Optional[FalkorDBGraph] = None,
+        relevance_score_fn: Optional[Callable[[float], float]] = None,
+        ssl: bool  = False,
+        pre_delete_collection: bool = False,
+    ) -> None:
+        try:
+            import falkordb 
+        except ImportError:
+            raise ImportError(
+                "Could not import falkordb python package."
+                "Please install it with `pip install falkordb`"
+            )
+        
+        # Only euclidean distance strategies is avaliable in falkor for now
+        if distance_strategy != DistanceStrategy.EUCLIDEAN_DISTANCE:
```

Comment:
from https://github.com/FalkorDB/FalkorDB/releases/tag/v4.2.0 cosine is exposed

Reply:
This is the [documentation](https://docs.falkordb.com/cypher/indexing.html)  I used and In that documentation, It says "At the moment the only supported similarity function is ‘euclidean’". Please update the docs.

Reply:
yes it was updated we added this